pallets/settlement/src/lib.rs

```diff
@@ -1164,15 +1164,30 @@ impl<T: Trait> Module<T> {
         <InstructionLegs<T>>::remove_prefix(instruction_id);
         <InstructionDetails<T>>::remove(instruction_id);
         <InstructionLegStatus<T>>::remove_prefix(instruction_id);
-        <InstructionAuthsPending>::remove(instruction_id);
-        <AuthsReceived>::remove_prefix(instruction_id);
-        // NB UserAuths mapping is not cleared
+        InstructionAuthsPending::remove(instruction_id);
+        AuthsReceived::remove_prefix(instruction_id);
+        Self::prune_user_auths(&legs, instruction_id);
+
         (
             instructions_processed,
             weight_for_execution.saturating_add(T::DbWeight::get().writes(5)),
         )
     }
 
+    fn prune_user_auths(legs: &Vec<(u64, Leg<T::Balance>)>, instruction_id: u64) {
+        // We remove duplicates in memory before triggering storage actions
+        let mut counter_parties = Vec::with_capacity(legs.len() * 2);
+        for (_id, leg) in legs {
```

Comment:
```suggestion
        for (_, leg) in legs {
```

Reply:
I prefer to name variables even if they are unused. They help in providing context.

Reply:
Context can be inferred from `legs`'s type. `_id` doesn't add anything functional here, so imo its just noise.

Reply:
I don't agree but I don't really care about this either. Will commit the change.